scripts/languageserver/provider_misc.jl

```diff
@@ -75,10 +75,36 @@ end
 abstract didChange <: Method
 
 function Respond(r::Request{didChange,DidChangeTextDocumentParams})
+        vtdi = r.params.textDocument
+        C = r.params.contentChanges
+        for c in C
+            c(vtdi)
+        end
     try
-        documents[r.params.textDocument.uri] = split(r.params.contentChanges[1].text, r"\r\n?|\n")
         return
     catch err
         return Response{didChange,Exception}("2.0",r.id,err)
     end
 end
+
+function (c::TextDocumentContentChangeEvent)(vtdi::VersionedTextDocumentIdentifier)
+    d = join(documents[vtdi.uri],'\n').data
```

Comment:
I think this is not very efficient. Why put everything together into one large string, if we can just modify those lines that are actually changed? I think the idea would be that most of these change requests probably only affect only one line, so those change requests can be handled very efficiently because you only need to touch one element in the array that holds the strings for a document.